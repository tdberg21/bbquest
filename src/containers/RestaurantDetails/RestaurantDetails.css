.restaurant-details-container {
  display: flex;
  justify-content: space-evenly;
  width: 101%;
  background-image: url(https://www.pigrigbbq.com/files/2017-02/background.jpg);
  height: 1500px;
  background-size: cover;
}

.reviews {
  margin: 2rem auto;
}

.reviews-section {
  background-color: rgb(243, 243, 243, .8);
  width: 40%;
  height: 55vh;
  padding: 0.83em;
  text-align: left;
  margin: 3em;
  border-radius: 0.75em;
}

.details-section {
  background-color: rgb(243, 243, 243, .8);
  width: 45%;
  height: 78vh;
  margin: 3em;
  border-radius: 0.75em;
}

.visited-button {
  display: inline-block;
  height: 2rem;
  background-color: red;
  color: #fff;
  border: none;
  padding: 0.5rem;
  font-size: 1em;
}

.details-lower-section {
  width: 60%;
  margin: 1em auto;
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  padding: 5px;
}

.details-label {
  font-weight: 800;
}