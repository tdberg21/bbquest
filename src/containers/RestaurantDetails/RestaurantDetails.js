import React, { Component } from 'react';
import { fetchDetails } from '../../helpers/apiCalls';
import { withRouter, Link } from 'react-router-dom';
import './RestaurantDetails.css';
import { connect } from 'react-redux';
import pig from '../../pig.gif';

export class RestaurantDetails extends Component {
  constructor () {
    super();
    
    this.state = {
      results: {
        reviews: [{ text: '' }, { text: '' }, { text: '' }]
      }
    };
  }

  async componentDidMount() {
    if (this.props.id) {
      let results = await fetchDetails(this.props.id);
      this.setState({ results });
    }
  }

  checkVisited = (id) => {
    this.props.history.push(`/restaurants/${this.props.name}/review`);
  }
  
  render () {
    if (this.state.results.reviews[0].text !== '') {
      return (
        <div className='restaurant-details-container'>
          <section className='details-section'>
            <h2>{this.props.name}</h2>
            <img src={this.props.image} alt="restaurant" height="200px" />
            <br/>
            <a className='yelp-link' href={this.props.url} target='_blank'>Visit on Yelp!</a>
            <p>Phone: {this.props.phone} </p>
            <p>Address: {this.props.address.address1},<br/> {this.props.address.city}, {this.props.address.state}</p>
            <p>Price Range: {this.props.price} </p>
            <p>Rating: {this.props.rating} /5</p>
            <button 
              className='visited-button'
              onClick={() => this.checkVisited(this.props.id)}
            >
              Mark As Visited
            </button>
          </section>
          <section className='reviews-section'>
            <h6>Reviews:</h6>
            <p className='reviews'> "{this.state.results.reviews[0].text}" </p>
            <p className='reviews'> "{this.state.results.reviews[1].text}" </p>
            <p className='reviews'> "{this.state.results.reviews[2].text}" </p>
            <Link to={'/restaurants/'}>Back</Link>
          </section>
        </div>
      );

    } else {
      return (
        <div className='restaurant-details-container'>
          <img src={pig} alt='loading gif' />
        </div>

      );
    }
    
  }
}

export const mapStateToProps = state => ({
  visited: state.visited
});

export default withRouter(connect(mapStateToProps)(RestaurantDetails));