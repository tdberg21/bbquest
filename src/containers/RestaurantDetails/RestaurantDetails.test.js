/*eslint-disable no-undef*/

import React from 'react';
import { RestaurantDetails, mapStateToProps } from './RestaurantDetails';
import { shallow } from 'enzyme';

describe('RESTAURANT DETAILS TESTS', () => {
  
  it('should match the snapshot without reviews', () => {
    const wrapper = shallow(<RestaurantDetails/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with reviews', () => {
    const mockResults = {reviews: ['review1', 'review2', 'review3']};
    const mockAddress = { address1: '1 Taco Street', city: 'tacotown', state: 'taco' };
    const wrapper = shallow(<RestaurantDetails address={mockAddress}/>);
    wrapper.setState({
      results: mockResults
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state with results as a key and a value of an array with 3 reviews with empty strings', () => {
    
    const wrapper = shallow(<RestaurantDetails />);
    const expected = { results: {reviews: [{ text: '' }, { text: '' }, { text: '' }]}};
    const result = wrapper.state();

    expect(result).toEqual(expected);
  });

  it('should call handleVisited on click of visited button', () => {
    const mockAddress = {address1:'1 Taco Street', city:'tacotown', state: 'taco'};
    const mockHistory = {push: jest.fn()};
    const wrapper = shallow(<RestaurantDetails address={mockAddress} checkVisited={jest.fn()} history={mockHistory}/>);
    const spy = jest.spyOn(wrapper.instance(), 'handleVisited');

    wrapper.setState({ results: { reviews: [{ text: 'taco' }, { text: 'taco' }, { text: 'taco' }]}});
    wrapper.find('.visited-button').simulate('click');

    expect(spy).toHaveBeenCalled();
  });

  it('when component is mounted, it should update state with results of fetchDetails call', async () => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve({
        reviews: [{ text: 'taco' }, { text: 'taco' }, { text: 'taco' }] 
      })
    }));
    const mockAddress = {address1: '123 way', city: 'tacoland', state: 'taco'};
    const wrapper = await shallow(<RestaurantDetails id='1' address={mockAddress} price='7' rating='5'/>);
    await wrapper.update();

    await expect(wrapper.state()).toEqual({results: { reviews: [{ text: '' }, { text: '' }, { text: '' }]}});
  });

  describe('mapStateToProps', () => {
    it('should return an object with the visited joints array', () => {
      const mockState = {
        visited: [{ name: 'PeeWees'}]
      };
      const expected = {
        visited: [{ name: 'PeeWees' }]
      };
      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    });
  });
});