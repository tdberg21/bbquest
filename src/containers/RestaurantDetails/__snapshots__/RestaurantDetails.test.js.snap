// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RESTAURANT DETAILS TESTS should match the snapshot with reviews 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantDetails
    address={
        Object {
            "address1": "1 Taco Street",
            "city": "tacotown",
            "state": "taco",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="details-section"
>
          <div
                    className="header-image-section"
          >
                    <h2 />
                    <img
                              alt="restaurant"
                              height="200px"
                              src={undefined}
                    />
                    <br />
          </div>
          <div
                    className="details-lower-section"
          >
                    <a
                              className="yelp-link"
                              href={undefined}
                              target="_blank"
                    >
                              Visit on Yelp!
                    </a>
                    <p>
                              <b>
                                        Phone: 
                              </b>
                    </p>
                    <p>
                              <b>
                                        Address: 
                              </b>
                              1 Taco Street
                              ,
                              <br />
                              tacotown
                              ,
                              taco
                    </p>
                    <p>
                              <b>
                                        Price Range: 
                              </b>
                    </p>
                    <p>
                              <b>
                                        Rating: 
                              </b>
                               /5
                    </p>
                    <button
                              className="visited-button"
                              onClick={[Function]}
                    >
                              Mark As Visited
                    </button>
          </div>
</section>,
        <section
          className="reviews-section"
>
          <h5>
                    Reviews: 
          </h5>
          <p
                    className="reviews"
          >
                    undefined
          </p>
          <p
                    className="reviews"
          >
                    undefined
          </p>
          <p
                    className="reviews"
          >
                    undefined
          </p>
          <Link
                    replace={false}
                    to="/restaurants/"
          >
                    Back
          </Link>
</section>,
      ],
      "className": "restaurant-details-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="header-image-section"
>
              <h2 />
              <img
                            alt="restaurant"
                            height="200px"
                            src={undefined}
              />
              <br />
</div>,
            <div
              className="details-lower-section"
>
              <a
                            className="yelp-link"
                            href={undefined}
                            target="_blank"
              >
                            Visit on Yelp!
              </a>
              <p>
                            <b>
                                          Phone: 
                            </b>
              </p>
              <p>
                            <b>
                                          Address: 
                            </b>
                            1 Taco Street
                            ,
                            <br />
                            tacotown
                            ,
                            taco
              </p>
              <p>
                            <b>
                                          Price Range: 
                            </b>
              </p>
              <p>
                            <b>
                                          Rating: 
                            </b>
                             /5
              </p>
              <button
                            className="visited-button"
                            onClick={[Function]}
              >
                            Mark As Visited
              </button>
</div>,
          ],
          "className": "details-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2 />,
                <img
                  alt="restaurant"
                  height="200px"
                  src={undefined}
/>,
                <br />,
              ],
              "className": "header-image-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "restaurant",
                  "height": "200px",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <a
                  className="yelp-link"
                  href={undefined}
                  target="_blank"
>
                  Visit on Yelp!
</a>,
                <p>
                  <b>
                                    Phone: 
                  </b>
</p>,
                <p>
                  <b>
                                    Address: 
                  </b>
                  1 Taco Street
                  ,
                  <br />
                  tacotown
                  ,
                  taco
</p>,
                <p>
                  <b>
                                    Price Range: 
                  </b>
</p>,
                <p>
                  <b>
                                    Rating: 
                  </b>
                   /5
</p>,
                <button
                  className="visited-button"
                  onClick={[Function]}
>
                  Mark As Visited
</button>,
              ],
              "className": "details-lower-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Visit on Yelp!",
                  "className": "yelp-link",
                  "href": undefined,
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Visit on Yelp!",
                "type": "a",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <b>
                      Phone: 
</b>,
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Phone: ",
                    },
                    "ref": null,
                    "rendered": "Phone: ",
                    "type": "b",
                  },
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <b>
                      Address: 
</b>,
                    "1 Taco Street",
                    ",",
                    <br />,
                    "tacotown",
                    ",",
                    "taco",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Address: ",
                    },
                    "ref": null,
                    "rendered": "Address: ",
                    "type": "b",
                  },
                  "1 Taco Street",
                  ",",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  "tacotown",
                  ",",
                  "taco",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <b>
                      Price Range: 
</b>,
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Price Range: ",
                    },
                    "ref": null,
                    "rendered": "Price Range: ",
                    "type": "b",
                  },
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <b>
                      Rating: 
</b>,
                    undefined,
                    " /5",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Rating: ",
                    },
                    "ref": null,
                    "rendered": "Rating: ",
                    "type": "b",
                  },
                  undefined,
                  " /5",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Mark As Visited",
                  "className": "visited-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Mark As Visited",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h5>
              Reviews: 
</h5>,
            <p
              className="reviews"
>
              undefined
</p>,
            <p
              className="reviews"
>
              undefined
</p>,
            <p
              className="reviews"
>
              undefined
</p>,
            <Link
              replace={false}
              to="/restaurants/"
>
              Back
</Link>,
          ],
          "className": "reviews-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reviews: ",
            },
            "ref": null,
            "rendered": "Reviews: ",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "undefined",
              "className": "reviews",
            },
            "ref": null,
            "rendered": "undefined",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "undefined",
              "className": "reviews",
            },
            "ref": null,
            "rendered": "undefined",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "undefined",
              "className": "reviews",
            },
            "ref": null,
            "rendered": "undefined",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Back",
              "replace": false,
              "to": "/restaurants/",
            },
            "ref": null,
            "rendered": "Back",
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="details-section"
>
            <div
                        className="header-image-section"
            >
                        <h2 />
                        <img
                                    alt="restaurant"
                                    height="200px"
                                    src={undefined}
                        />
                        <br />
            </div>
            <div
                        className="details-lower-section"
            >
                        <a
                                    className="yelp-link"
                                    href={undefined}
                                    target="_blank"
                        >
                                    Visit on Yelp!
                        </a>
                        <p>
                                    <b>
                                                Phone: 
                                    </b>
                        </p>
                        <p>
                                    <b>
                                                Address: 
                                    </b>
                                    1 Taco Street
                                    ,
                                    <br />
                                    tacotown
                                    ,
                                    taco
                        </p>
                        <p>
                                    <b>
                                                Price Range: 
                                    </b>
                        </p>
                        <p>
                                    <b>
                                                Rating: 
                                    </b>
                                     /5
                        </p>
                        <button
                                    className="visited-button"
                                    onClick={[Function]}
                        >
                                    Mark As Visited
                        </button>
            </div>
</section>,
          <section
            className="reviews-section"
>
            <h5>
                        Reviews: 
            </h5>
            <p
                        className="reviews"
            >
                        undefined
            </p>
            <p
                        className="reviews"
            >
                        undefined
            </p>
            <p
                        className="reviews"
            >
                        undefined
            </p>
            <Link
                        replace={false}
                        to="/restaurants/"
            >
                        Back
            </Link>
</section>,
        ],
        "className": "restaurant-details-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="header-image-section"
>
                <h2 />
                <img
                                alt="restaurant"
                                height="200px"
                                src={undefined}
                />
                <br />
</div>,
              <div
                className="details-lower-section"
>
                <a
                                className="yelp-link"
                                href={undefined}
                                target="_blank"
                >
                                Visit on Yelp!
                </a>
                <p>
                                <b>
                                                Phone: 
                                </b>
                </p>
                <p>
                                <b>
                                                Address: 
                                </b>
                                1 Taco Street
                                ,
                                <br />
                                tacotown
                                ,
                                taco
                </p>
                <p>
                                <b>
                                                Price Range: 
                                </b>
                </p>
                <p>
                                <b>
                                                Rating: 
                                </b>
                                 /5
                </p>
                <button
                                className="visited-button"
                                onClick={[Function]}
                >
                                Mark As Visited
                </button>
</div>,
            ],
            "className": "details-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2 />,
                  <img
                    alt="restaurant"
                    height="200px"
                    src={undefined}
/>,
                  <br />,
                ],
                "className": "header-image-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "restaurant",
                    "height": "200px",
                    "src": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    className="yelp-link"
                    href={undefined}
                    target="_blank"
>
                    Visit on Yelp!
</a>,
                  <p>
                    <b>
                                        Phone: 
                    </b>
</p>,
                  <p>
                    <b>
                                        Address: 
                    </b>
                    1 Taco Street
                    ,
                    <br />
                    tacotown
                    ,
                    taco
</p>,
                  <p>
                    <b>
                                        Price Range: 
                    </b>
</p>,
                  <p>
                    <b>
                                        Rating: 
                    </b>
                     /5
</p>,
                  <button
                    className="visited-button"
                    onClick={[Function]}
>
                    Mark As Visited
</button>,
                ],
                "className": "details-lower-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Visit on Yelp!",
                    "className": "yelp-link",
                    "href": undefined,
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "Visit on Yelp!",
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <b>
                        Phone: 
</b>,
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Phone: ",
                      },
                      "ref": null,
                      "rendered": "Phone: ",
                      "type": "b",
                    },
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <b>
                        Address: 
</b>,
                      "1 Taco Street",
                      ",",
                      <br />,
                      "tacotown",
                      ",",
                      "taco",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Address: ",
                      },
                      "ref": null,
                      "rendered": "Address: ",
                      "type": "b",
                    },
                    "1 Taco Street",
                    ",",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    "tacotown",
                    ",",
                    "taco",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <b>
                        Price Range: 
</b>,
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Price Range: ",
                      },
                      "ref": null,
                      "rendered": "Price Range: ",
                      "type": "b",
                    },
                    undefined,
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <b>
                        Rating: 
</b>,
                      undefined,
                      " /5",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Rating: ",
                      },
                      "ref": null,
                      "rendered": "Rating: ",
                      "type": "b",
                    },
                    undefined,
                    " /5",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Mark As Visited",
                    "className": "visited-button",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Mark As Visited",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5>
                Reviews: 
</h5>,
              <p
                className="reviews"
>
                undefined
</p>,
              <p
                className="reviews"
>
                undefined
</p>,
              <p
                className="reviews"
>
                undefined
</p>,
              <Link
                replace={false}
                to="/restaurants/"
>
                Back
</Link>,
            ],
            "className": "reviews-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reviews: ",
              },
              "ref": null,
              "rendered": "Reviews: ",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "undefined",
                "className": "reviews",
              },
              "ref": null,
              "rendered": "undefined",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "undefined",
                "className": "reviews",
              },
              "ref": null,
              "rendered": "undefined",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "undefined",
                "className": "reviews",
              },
              "ref": null,
              "rendered": "undefined",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Back",
                "replace": false,
                "to": "/restaurants/",
              },
              "ref": null,
              "rendered": "Back",
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`RESTAURANT DETAILS TESTS should match the snapshot without reviews 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantDetails />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="loading gif"
        className="pig-gif"
        src="pig.gif"
/>,
      "className": "restaurant-details-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "loading gif",
        "className": "pig-gif",
        "src": "pig.gif",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="loading gif"
          className="pig-gif"
          src="pig.gif"
/>,
        "className": "restaurant-details-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "loading gif",
          "className": "pig-gif",
          "src": "pig.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
